app/services/credentials_verifier.go

```diff
@@ -45,5 +47,16 @@ func CredentialsVerifier(store data.AccountStore, cfg *app.Config, username stri
 		return nil, FieldErrors{{"credentials", ErrExpired}}
 	}
 
+	//Check TOTP MFA
+	if account.TOTPEnabled() {
+		secret, err := compat.Decrypt([]byte(account.TOTPSecret.String), cfg.DBEncryptionKey)
+		if err != nil {
+			return nil, errors.Wrap(err, "TOTPDecrypt")
+		}
+		if !totp.Validate(totpCode, secret) {
+			return nil, FieldErrors{{"totp", ErrInvalidOrExpired}}
```

Comment:
@xcskier56 you asked how a client could know that someone needs to supply a token. i expect that it will be a guess-and-test process, with this error signaling the extra requirement.

1. collect and submit username/password
2. catch totp error
3. reveal otp input and merge with username/password in a second submit

Reply:
This may have come up in previous discussion, but if other authentication factors are on the roadmap, it may make sense to have an explicit API to signal required factors. 

I also worry that coupling TOTP and basic credentials into the same call makes OIDC ACR-like "step-up" authentication more difficult (certain endpoints only require credentialed authn, but others may require elevated factors). That may be out of scope for Keratin; i.e. Keratin is one among many identity providers that can satisfy factors, not the application that assembles and verifies them.

Reply:
> it may make sense to have an explicit API to signal required factors

Makes sense to me. That seems like an easy addition. Do you think it would require changes at this step?

> "step-up" authentication more difficult (certain endpoints only require credentialed authn, but others may require elevated factors)

I've been assuming Keratin would need to handle this for installations where 2FA becomes mandated. I hadn't considered exposing the half-authenticated state to the host application, but I think it'd be an easy addition with a new JWT claim. Thoughts?